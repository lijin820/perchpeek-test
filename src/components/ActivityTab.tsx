import React, { ReactElement } from "react";
import { Button } from "antd";
import { PlusOutlined } from "@ant-design/icons";

function ActivityTab(): ReactElement {
  return (
    <div className="Layout">
      <Button type="primary" icon={<PlusOutlined />}>
        Add new activity
      </Button>
    </div>
  );
}

export default ActivityTab;
