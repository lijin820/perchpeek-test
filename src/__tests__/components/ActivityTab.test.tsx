import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import ActivityTab from "../../components/ActivityTab";

describe("ActivityTab", () => {
  const initialState = { activities: [] };
  const mockStore = configureStore();
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let store: any;

  describe("rendering", () => {
    store = mockStore(initialState);
    let wrapper: ShallowWrapper;
    beforeEach(() => {
      wrapper = shallow(
        <Provider store={store}>
          <ActivityTab />
        </Provider>
      );
    });

    it("renders the component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("renders parent component", () => {
      expect(wrapper.find(ActivityTab)).toHaveLength(1);
    });
  });
});
