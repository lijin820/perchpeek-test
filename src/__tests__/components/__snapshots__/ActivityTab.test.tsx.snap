// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityTab rendering renders the component 1`] = `
<div>
  <Button
    data-id="add-button"
    icon={<ForwardRef(PlusOutlined) />}
    onClick={[Function]}
    type="primary"
  >
    Add new activity
  </Button>
  <AddActivityModal
    data-id="add-activity-modal"
    setVisible={[Function]}
    visible={false}
  />
  <ActivityList
    activities={Array []}
  />
</div>
`;
